### ISSUE TITLE
能否分享一下最终训练完的权重？

### ISSUE THREAD

**Issue by isunLt** [PKUHaoWang/EmbodiedOcc2#3]
我按照ReadMe中的指令尝试在OccScannet-Mini和EmbodiedOcc-ScanNet-Mini上复现论文中的结果，但实际上我在OccScannet-Mini只跑到了45.66 mIoU，如下图所示
<img width="2994" height="782" alt="Image" src="https://github.com/user-attachments/assets/4181c030-0e89-445b-8c08-69c3290622df" />
在EmbodiedOcc-ScanNet-Mini上只跑到了40.65 mIoU，请问是不是我有哪里出现了错误？
<img width="2975" height="1338" alt="Image" src="https://github.com/user-attachments/assets/0d88bd06-d139-4765-b1d3-6869eb98c73b" />
我将训练log和weight分享在以下链接
https://drive.google.com/drive/folders/1l8sw6PIPFXfYaWZtIw6YHZxZypfZbOtR?usp=sharing

**Comment by PKUHaoWang** [comment]
您好，我在 OccScannet-Mini 上的 Log 如下

<img width="1139" height="199" alt="Image" src="https://github.com/user-attachments/assets/aeac764c-38f6-4a0d-baf8-34d7a52665ae" />


EmbodiedOcc-ScanNet-Mini 上的 Log 如下

<img width="1125" height="352" alt="Image" src="https://github.com/user-attachments/assets/84c78ce5-7b3d-4a72-a62e-ab464d160c41" />

需要注意的是，在 EmbodiedOcc-ScanNet-Mini 上训练后，需要单独使用单卡进行测试，多卡的测试结果会有bug出现导致最终结果计算存在问题。

最终权重我将会在整理好后发出

**Comment by isunLt** [comment]
> 您好，我在 OccScannet-Mini 上的 Log 如下
> 
> <img alt="Image" width="1139" height="199" src="https://private-user-images.githubusercontent.com/89617851/497517725-aeac764c-38f6-4a0d-baf8-34d7a52665ae.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk2NDUyNDAsIm5iZiI6MTc1OTY0NDk0MCwicGF0aCI6Ii84OTYxNzg1MS80OTc1MTc3MjUtYWVhYzc2NGMtMzhmNi00YTBkLWJhZjgtMzRkN2E1MjY2NWFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA1VDA2MTU0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwM2U4OTc1YjZjZWM2ODAzN2ZiM2MxM2E3ZjEwMjhlMjIyNzI3MDc5ODk2MDQ1YjE3YzI2NWQ1OTNjMmY1YTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.V61vjGj6C0Qda1yHcjVAO2qFvII1Ic-OpFvYatNcW7k">
> EmbodiedOcc-ScanNet-Mini 上的 Log 如下
> 
> <img alt="Image" width="1125" height="352" src="https://private-user-images.githubusercontent.com/89617851/497518420-84c78ce5-7b3d-4a72-a62e-ab464d160c41.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk2NDUyNDAsIm5iZiI6MTc1OTY0NDk0MCwicGF0aCI6Ii84OTYxNzg1MS80OTc1MTg0MjAtODRjNzhjZTUtN2IzZC00YTcyLWE2MmUtYWI0NjRkMTYwYzQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA1VDA2MTU0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwYmUwMDVhYjIzNDkxOTI4ZmRmZjk5MTliNTRmNWJhZDg0MjRmMDFiOWE3YTViMmJjMGE0YjAyMzlhODQ0MmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0yGY1KyjHA42vymiANRHcgxZDALZMYeKVU5dRxM265A">
> 需要注意的是，在 EmbodiedOcc-ScanNet-Mini 上训练后，需要单独使用单卡进行测试，多卡的测试结果会有bug出现导致最终结果计算存在问题。
> 
> 最终权重我将会在整理好后发出

感谢你的回复，期待你的最终权重。

**Comment by PKUHaoWang** [comment]
您可以先使用我们在 https://huggingface.co/Hao-Wang/EmbodiedOcc2/tree/main 发布的模型权重进行验证。后面我们会更新此链接。

**Comment by Lcp9202** [comment]
> 您可以先使用我们在 https://huggingface.co/Hao-Wang/EmbodiedOcc2/tree/main 发布的模型权重进行验证。后面我们会更新此链接。

### 🧩 Issue Description

I used the pretrained checkpoint **`Occ-ScanNet-mini.pth`** released on Hugging Face together with the configuration file **`train_mono_mini_geo_config.py`** and the visualization script **`vis_mono.py`**.
When examining the visualization routine, I noticed that the final visualization mask is constructed as the intersection of **non-empty ground-truth voxels**, **non-empty predicted voxels**, **non-unknown predicted voxels**, and the **`fov_mask`**.

However, when visualizing with this mask, I observed only a very small number of visible points.
To further investigate, I experimented with different masking combinations as follows:

---

#### 1️⃣ (predicted ≠ empty) ∧ (predicted ≠ unknown) ∧ fov mask

<img width="988" height="761" alt="pred≠empty_pred≠unk_fov" src="https://github.com/user-attachments/assets/eafd973b-4daf-4fef-8847-1bfcc7a352bc" />

#### 2️⃣ (GT ≠ empty) ∧ (predicted ≠ empty) ∧ (predicted ≠ unknown)

<img width="957" height="671" alt="GT≠empty_pred≠empty_pred≠unk" src="https://github.com/user-attachments/assets/922b4242-846c-4221-abf8-165436c06540" />

#### 3️⃣ (predicted ≠ empty) ∧ (predicted ≠ unknown)

<img width="861" height="785" alt="pred≠empty_pred≠unk" src="https://github.com/user-attachments/assets/253d53b2-31cc-4442-84db-59b2f858056c" />

---

In all of these cases, the visualization results appear severely degraded —
it is difficult to identify any meaningful geometric structure.
Labels that should correspond to the *floor* (green) or *ceiling* (red) are randomly scattered in 3D space.

It is worth noting that I did **not** modify any configuration parameters, nor did I fine-tune the provided checkpoint.

---

### ❓ Question

Could you please clarify why this behavior occurs?
Is there any additional preprocessing, coordinate transformation, or masking step required to correctly reproduce the expected geometric layout during visualization?

**Comment by Lcp9202** [comment]
> 您可以先使用我们在 https://huggingface.co/Hao-Wang/EmbodiedOcc2/tree/main 发布的模型权重进行验证。后面我们会更新此链接。

I’m trying to visualize with the following setup:

Config: Occ-ScanNet-mini/train_mono_mini_geo_config.py

Checkpoint: Occ-ScanNet-mini/Occ-ScanNet-mini.pth (downloaded from your HuggingFace link)

However, during model loading, a large number of missing_keys and unexpected_keys were reported, indicating that the checkpoint’s architecture might not match the model built from this config.

In contrast, I noticed that your vis_mono.py script by default uses:

Config: mini-dataset-product/train_mono_mini_exp_config.py

Checkpoint: mini-dataset-product/epoch_20.pth

Could you please clarify:

Should I use the same mini-dataset-product config and weight for visualization?

Or is the HuggingFace-released Occ-ScanNet-mini.pth corresponding to a different config structure?

**Comment by Leooncode** [comment]
> 我按照ReadMe中的指令尝试在OccScannet-Mini和EmbodiedOcc-ScanNet-Mini上复现论文中的结果，但实际上我在OccScannet-Mini只跑到了45.66 mIoU，如下图所示 <img alt="Image" width="2000" height="782" src="https://private-user-images.githubusercontent.com/37700210/485953716-4181c030-0e89-445b-8c08-69c3290622df.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc2NjY4OTIsIm5iZiI6MTc2NzY2NjU5MiwicGF0aCI6Ii8zNzcwMDIxMC80ODU5NTM3MTYtNDE4MWMwMzAtMGU4OS00NDViLThjMDgtNjljMzI5MDYyMmRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA2VDAyMjk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmYjhmMDg1NWMyMTIxN2EzMzNhMDk4ZDIwZWU3MzU2Y2IxOTA2MGFjY2Y4ZDlmMjhmMzBmMTcyMjVmZDkzZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lMQ7zO4vYJymQgrqM68Ik7c_jYcm2U1uC0F7iIzRwwo"> 在EmbodiedOcc-ScanNet-Mini上只跑到了40.65 mIoU，请问是不是我有哪里出现了错误？ <img alt="Image" width="2000" height="1338" src="https://private-user-images.githubusercontent.com/37700210/485954418-0d88bd06-d139-4765-b1d3-6869eb98c73b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc2NjY4OTIsIm5iZiI6MTc2NzY2NjU5MiwicGF0aCI6Ii8zNzcwMDIxMC80ODU5NTQ0MTgtMGQ4OGJkMDYtZDEzOS00NzY1LWIxZDMtNjg2OWViOThjNzNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA2VDAyMjk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyNWVhNWQ0ZmVkNWIzZDIyZWM5ZjY3Nzk4ZTU2OGY0ZTQ5YTJjZTdiZWQwOTM0YzQyZjkxMzNjZDliYWJkZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.03RVXwXbeHy1cl6KziAELbYBoG_fWB0euEKO85ERbXQ"> 我将训练log和weight分享在以下链接 https://drive.google.com/drive/folders/1l8sw6PIPFXfYaWZtIw6YHZxZypfZbOtR?usp=sharing

你好，我复现出来的结果也和你类似，请问解决了吗？
